src/PhpSpec/Listener/MethodNotFoundListener.php

```diff
@@ -27,12 +28,28 @@ class MethodNotFoundListener implements EventSubscriberInterface
     private $resources;
     private $generator;
     private $methods = array();
+    private $wrongMethodNames = array();
+    /**
+     * @var VoterInterface
+     */
+    private $nameChecker;
 
-    public function __construct(IO $io, ResourceManagerInterface $resources, GeneratorManager $generator)
-    {
+    /**
+     * @param IO $io
+     * @param ResourceManagerInterface $resources
+     * @param GeneratorManager $generator
+     * @param VoterInterface $nameChecker
+     */
+    public function __construct(
+        IO $io,
+        ResourceManagerInterface $resources,
+        GeneratorManager $generator,
+        VoterInterface $nameChecker
```

Comment:
Changing the constructor in this way is a BC break.

Please make this optional and instantiate a new copy inside the constructor if it's not provided (we can remove this in PhpSpec 3.0)